jest/private/jest_test.bzl

```diff
@@ -1,18 +1,28 @@
 "Implementation details for jest_test rule"
 
 load("@aspect_rules_js//js:libs.bzl", "js_binary_lib", "js_lib_helpers")
-load("@aspect_bazel_lib//lib:paths.bzl", "relative_file")
+load("@aspect_bazel_lib//lib:paths.bzl", "to_manifest_path", "to_workspace_path")
 load("@bazel_skylib//lib:dicts.bzl", "dicts")
 
 _attrs = dicts.add(js_binary_lib.attrs, {
+    "auto_configure_reporters": attr.bool(
```

Comment:
I think its better to be specific with this flag and name it `auto_configure_jest_junit_reporter` so if we have more reporters auto-configured in the future they can be opted out individually

Reply:
What is the "default" reporter doing? Should that one be always on without an opt-out?

Reply:
Using `--reporters` on the command line overrides any reporters they set. They don't get merged with the config file. If we don't declare the `default` reporter, then we get no stdout output to the regular bazel test log. So calling it `auto_configure_jest_junit_reporter ` wouldn't quite be accurate.

Reply:
Ahh. So sounds like the default reporter should be on no matter what and the `auto_configure_jest_junit_reporter` attribute should just control the junit one?

Reply:
Thing is, if we always set the default reporter using `--reporters` then their custom ones won't work. So we have to set both, or none.

Reply:
If we don't set `default`, then we just get the xml test logs but not the regular bazel ones that (I think) just capture stdout.

Reply:
It looks like `--reporters` is additive since its specified twice in these args. Doesn't replace the reporters in the user config but just adds to the list?

Reply:
I think the presence of a `--reporters` on the cli causes it to ignore any in the config, but let me double check that again.

Reply:
I just checked again. So if I add:
```
reporters: ["default"]
```
to a jest config file, then remove `--reporters default` from our rule, but keep the junit `--reporter`, then the test only gets unit xml output, no default reporting and the non-xml bazel test log doesn't get any reporting. So based on that, I'm assuming that the presence of any cli `--reporter` nullifies any reporters in the config file.

Reply:
If we wanted finer grained control, we could go back to the approach where we generate a new config file that loads and extends the user config file.

Reply:
Oh... that is surprising and unfortunate behavior.

Puts users in a bit of a pickle since if they turn off `auto_configure_reporters` then they loose the `JEST_JUNIT_OUTPUT_FILE`.

Additive would be ideal but there is no easy way to do that without generating an extended config so I think next best is to document the behaviour that when True, the auto-conf reporters will override the jest config reporters and note that in the main rule docstring that that is the default behavior.

I suppose we should also always set `JEST_JUNIT_OUTPUT_FILE` so if the user decides to set `auto_configure_reporters` to False they can still add their own `jest-junit` reporter and the output path will still be what Bazel expects.

Reply:
Yeah, that sounds good.